### PR TITLE
Add repos endpoint

### PR DESCRIPTION
**Pull request by alice-sawatzky**
add an endpoint at `/api/repos` to list repos

- [x] figure out routing??
- [x] add handler
- [x] add tests
- [x] document endpoint

Relies on https://github.com/chartmuseum/storage/pull/48, so once that merges we should revert https://github.com/helm/chartmuseum/pull/364/commits/5ca02f608563175cc4e9dbbf624f8f28c0aa1b34

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @alice-sawatzky isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by alice-sawatzky** [comment]
@thelinuxfoundation try again?

**Comment by alice-sawatzky** [comment]
I signed it

**Comment by alice-sawatzky** [comment]
@thelinuxfoundation

**Comment by alice-sawatzky** [comment]
retrigger

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ go 1.14
 replace (
 	github.com/NetEase-Object-Storage/nos-golang-sdk => github.com/karuppiah7890/nos-golang-sdk v0.0.0-20191116042345-0792ba35abcc
 	go.etcd.io/etcd => github.com/eddycjy/etcd v0.5.0-alpha.5.0.20200218102753-4258cdd2efdf
+	github.com/chartmuseum/storage => github.com/farmersedgeinc/storage v0.9.1-0.20200810211930-42f6c50507a1
```
Comment (jdolitsky) [inline go.mod:8]: Can we remove this? The chartmuseum 'main' branch now has your changes

File: README.md [hunk 2]
```diff
@@ -29,6 +29,10 @@ Powered by some great Go technology:
 - `GET /charts/mychart-0.1.0.tgz` - retrieved when you run `helm install chartmuseum/mychart`
 - `GET /charts/mychart-0.1.0.tgz.prov` - retrieved when you run `helm install` with the `--verify` flag
 
+### Repository Manipulation
+- `GET /api/repos` - list chart repos (only works with multitenancy enabled).
+- `GET /api/<fragment>/repos` - list sub-repos within a repository (only works with multitenancy enabled).
+
```
Comment (jdolitsky) [inline README.md:35]: Can we list this underneath "Chart Manipulation"?

**Review by jdolitsky — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (jdolitsky) [reply go.mod:8]: whoops - my apologies, has not been yet

**Comment by jdolitsky** [comment]
Since this is a large feature, going to slate it for v0.14 so we can release a v0.13. Thanks again for working on this

**Comment by alice-sawatzky** [comment]
no problem! I was learning go while writing this, so please let me know if there's any issues i've missed and i'll be happy to fix (and learn from it!)

File: go.mod (hunk 1 above)
Comment (alice-sawatzky) [reply go.mod:8]: just bumping: reviewing the storage library changes will be a blocker for getting this merged. idk when you're planning your next release so maybe that's not an issue yet

**Comment by jdolitsky** [comment]
closing as outdated

**Pull request closed without merging**